frontend/src/components/pages/topics/Tab.Messages/index.tsx

```diff
@@ -1688,38 +1716,66 @@ class MessageSearchFilterBar extends Component<{ messageSearch: MessageSearch, c
+                                                this.currentFilter!.name = e.target.value;
+                                                this.hasChanges = true;
+                                            }}
+                                            placeholder="This name will appear in the filter bar"
+                                    />
+                                </FormField>}
+                            </GridItem>
+                            <GridItem />
+                            <GridItem display="flex" gap={4} flexDirection="column">
+                                {this.currentFilter &&
+                                    <FormField
+                                        label="Filter code"
+                                    >
+                                        <Box borderRadius={20}>
+                                        <FilterEditor
+                                            value={this.currentFilter!.code}
+                                            onValueChange={(code, transpiled) => {
+                                                this.currentFilter!.code = code;
+                                                this.currentFilter!.transpiledCode = transpiled;
+                                                this.hasChanges = true;
+                                            }}
+                                        />
+                                        </Box>
+                                    </FormField>
+                                }
+
+                                <UnorderedList>
+                                    <ListItem>return true allows messages, return false discards them.</ListItem>
+                                    <ListItem>Available params are offset and partitionID (number), key (string), value, and headers (objects)</ListItem>
+                                    <ListItem>Multiple active filters are combined with 'and'.</ListItem>
+                                    <ListItem>The context is re-used between messages, but every partition has its own context</ListItem>
+                                </UnorderedList>
+
+
+                            </GridItem>
+                            <GridItem>
+                                <Heading mb={6}>Examples</Heading>
+                                <UnorderedList styleType="none" margin={0} spacing={4}>
+                                    <ListItem><Code>value != null</Code> skips tombstone messages</ListItem>
+                                    <ListItem><Code>if (key == 'example') return true</Code>
+                                        only returns messages where keys equal <Code>'example'</Code></ListItem>
+                                    <ListItem><Code>{'headers.myVersionHeader && (headers.myVersionHeader >= 2)'}</Code> only returns messages that have a header entry like <Code>{'{key: "myVersionHeader", "value:" 12345}'}</Code></ListItem>
```

Comment:
```suggestion
```

I think that's not working at the moment until we re-introduce header parsing again